src/Ocelot/LoadBalancer/Middleware/LoadBalancingMiddleware.cs

```diff
@@ -38,7 +42,9 @@ public async Task Invoke(DownstreamContext context)
                 return;
             }
 
-            context.DownstreamRequest.Host = hostAndPort.Data.DownstreamHost;
+            var baseHostInfo = _baseHostProvider.GetBaseHostInfo(hostAndPort.Data.DownstreamHost);
```

Comment:
Yes, if we can return application name as field in DownstreamHost object, that will eliminate the need of parsing the incoming host address. This will still need a field on DownstreamRequest object to hold the value.

Reply:
This is what we should do, add a path field to the DownstreamHostAndPort object (maybe rename this). Then in the service discovery provider code set path if it exists.

Reply:
Sounds good.